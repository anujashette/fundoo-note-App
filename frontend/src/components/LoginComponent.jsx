import React, { Component } from 'react';
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent';
import { ValidatorForm, TextValidator } from 'react-material-ui-form-validator';
import { login } from '../services/UserService'
import Snackbar from '@material-ui/core/Snackbar'
import Typography from '@material-ui/core/Typography';
import { Redirect } from 'react-router-dom'
import '../App.css'

class LoginComponent extends Component {
    constructor(props) {
        super(props);
        this.state = {
            formData: {
                email: '',
                password: '',


            },
            snackbaropen: false,
            snackbarmsg: '',
            submitted: false,
            loggedIn: false
        }
        this.handleSubmit = this.handleSubmit.bind(this)
        this.handleRegister = this.handleRegister.bind(this)

    }

    snackBarClose = () => {
        this.setState({ snackbaropen: false })
    }
    handleChange = event => {
        const { formData } = this.state;
        formData[event.target.name] = event.target.value;
        this.setState({ formData });
        // console.log('form data',this.state.formData)
    }

    handleSubmit(event) {
        event.preventDefault();
        console.log('form data in submit', this.state.formData)

        if (!this.state.formData.email) {
            console.log('email should not be empty')
            this.setState({ snackbaropen: true, snackbarmsg: 'Email should not be empty' })
        }
        else if (this.state.formData.password.length < 6) {
            this.setState({ snackbaropen: true, snackbarmsg: 'password too short. At least 6 characters' })
        }
        else {
            let userInput = {
                email: this.state.formData.email,
                password: this.state.formData.password
            }
            console.log('form data in submit', userInput)

            // var formData = Object.assign({}, this.state.formData);
            // Object.keys(formData).map((key, index) => {
            //     formData[key] = '';
            //     return ''
            // });
            // this.setState({ formData })

            login(userInput)
                .then(async (response) => {
                    // if(response.data.data.email === this.state.formData.email  && response.data.data.password === this.state.formData.password ){
                    console.log('login', response.data.data)
                    localStorage.setItem('LoginToken', response.data.data.token)
                    localStorage.setItem('firstname', response.data.data.user.firstname)
                    localStorage.setItem('lastname', response.data.data.user.lastname)
                    localStorage.setItem('email', response.data.data.user.email)
                    localStorage.setItem('imageurl', response.data.data.user.imageurl)
                    localStorage.setItem('token1', true)

                    await this.setState({ snackbaropen: true, snackbarmsg: response.data.message, loggedIn: true })
                    console.log("in login", this.state.loggedIn);
                    //  this.renderRedirect()
                    // if(localStorage.getItem('token1') !== undefined)
                    // this.props.props.history.push('/dashboard')
                    // else
                    // this.props.props.history.push('/')
                    // }
                }).catch((errorMessages) => {
                    console.log('login error', errorMessages)
                    this.setState({ snackbaropen: true, snackbarmsg: 'Email is not verified OR Credentials are incorrect' })
                })
        }
    }
    renderRedirect = () => {
        console.log("in rediect", this.state.loggedIn);

        if (this.state.loggedIn) {
            return <Redirect to={{ pathname: '/dashboard' }} />
            // this.props.props.history.push('/dashboard')
        }
    }

    handleRegister = () => {
        console.log('in handle register')
        this.props.props.history.push('/registration')
    }

    handleForgetPassword = () => {
        this.props.props.history.push('/forgotpass')
    }

    localStorageClear = () =>{
        localStorage.clear()
    }

    render() {
        return (
            this.state.loggedIn ?
                (<Redirect
                    to={{
                        pathname: "/dashboard",
                        // state: { from: props.location }
                    }}
                />
                ) : (
                    <div>

                        <Snackbar
                            anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
                            open={this.state.snackbaropen}
                            autoHideDuration={5000}
                            onClose={this.snackBarClose}
                            message={<span id="message-id">{this.state.snackbarmsg}</span>}
                        ></Snackbar>

                        <Card id="cardidlogin" >
                            <CardContent>
                                <Typography variant="h5">
                                    <span className="font-color">F</span>
                                    <span className="font-color">u</span>
                                    <span className="font-color">n</span>
                                    <span className="font-color">d</span>
                                    <span className="font-color">o</span>
                                    <span className="font-color">o</span>
                                </Typography>
                                <Typography variant="h6" component="h2">
                                    Sign in
                        </Typography>

                                <br />
                                {this.localStorageClear}
                                <ValidatorForm
                                    autoComplete="off"
                                    ref="form"
                                    onSubmit={this.handleSubmit}
                                >
                                    <br />
                                    <TextValidator
                                        label="Email"
                                        onChange={this.handleChange}
                                        name="email"
                                        value={this.state.formData.email}
                                        className="Emaillogin"
                                        validators={['required', 'isEmail']}
                                        errorMessages={['this field is required', 'email is not valid']}
                                    />

                                    <br />

                                    <TextValidator
                                        label="Password"
                                        onChange={this.handleChange}
                                        name="password"
                                        type='password'
                                        value={this.state.formData.password}
                                        validators={['required']}
                                        className="Password"
                                        errorMessages={['this field is required']}
                                    />
                                    <br />
                                    <div className="CreateAccount">
                                        <button
                                            className="login-button-letter"
                                            onClick={this.handleForgetPassword}
                                        >
                                            Forgot Password?
                                </button>
                                    </div>
                                    <br />

                                    <div className="LoginButton">
                                        <button
                                            className="login-button-letter"
                                            onClick={this.handleRegister}>
                                            Create account
                                </button>
                                        <button
                                            type="submit"
                                            className="button-letter"
                                        >
                                            Login
                                </button>
                                    </div>
                                </ValidatorForm>
                            </CardContent>
                        </Card>
                    </div>
                )
        );
    }
}

export default LoginComponent;