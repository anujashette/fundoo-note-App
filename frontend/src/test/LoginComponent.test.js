import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, configure } from 'enzyme';
import LoginComponent from '../components/LoginComponent';
configure({ adapter: new Adapter() });

describe("LoginComponent", () => {
    it("should render LoginComponent", () => {
        const wrapper = shallow(<LoginComponent />);
        expect(wrapper.exists()).toBe(true)
    });

    it("Should validated input fields", () => {
        const wrapper = shallow(<LoginComponent />)
        let prevented = false
        wrapper.find("ValidatorForm").simulate("submit", {
            preventDefault: () => {
                prevented = true
            }
        });
        expect(prevented).toBe(true)

    });

    it("Should validated input fields", () => {
        const wrapper = shallow(<LoginComponent />)
        expect(wrapper.getElements()).toMatchSnapshot();
    });
    // it("onChange param is the same value as the input element's value property", () => {
    //     const wrapper = shallow(<LoginComponent />)

    //     wrapper.find('TextValidator').simulate('change', { target: { value: 'matched' } });
    //     expect('email').toBe('matched');
    // });
});