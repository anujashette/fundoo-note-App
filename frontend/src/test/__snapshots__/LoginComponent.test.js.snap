// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent Should validated input fields 1`] = `
Array [
  <div>
    <ForwardRef(WithStyles)
      anchorOrigin={
        Object {
          "horizontal": "center",
          "vertical": "top",
        }
      }
      autoHideDuration={5000}
      message={
        <span
          id="message-id"
        >
          
        </span>
      }
      onClose={[Function]}
      open={false}
    />
    <ForwardRef(WithStyles)
      id="cardidlogin"
    >
      <ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          variant="h5"
        >
          <span
            className="font-color"
          >
            F
          </span>
          <span
            className="font-color"
          >
            u
          </span>
          <span
            className="font-color"
          >
            n
          </span>
          <span
            className="font-color"
          >
            d
          </span>
          <span
            className="font-color"
          >
            o
          </span>
          <span
            className="font-color"
          >
            o
          </span>
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          component="h2"
          variant="h6"
        >
          Sign in
        </ForwardRef(WithStyles)>
        <br />
        [Function]
        <ValidatorForm
          autoComplete="off"
          debounceTime={0}
          onError={[Function]}
          onSubmit={[Function]}
        >
          <br />
          <TextValidator
            className="Emaillogin"
            errorMessages={
              Array [
                "this field is required",
                "email is not valid",
              ]
            }
            label="Email"
            name="email"
            onChange={[Function]}
            validatorListener={[Function]}
            validators={
              Array [
                "required",
                "isEmail",
              ]
            }
            value=""
          />
          <br />
          <TextValidator
            className="Password"
            errorMessages={
              Array [
                "this field is required",
              ]
            }
            label="Password"
            name="password"
            onChange={[Function]}
            type="password"
            validatorListener={[Function]}
            validators={
              Array [
                "required",
              ]
            }
            value=""
          />
          <br />
          <div
            className="CreateAccount"
          >
            <button
              className="login-button-letter"
              onClick={[Function]}
            >
              Forgot Password?
            </button>
          </div>
          <br />
          <div
            className="LoginButton"
          >
            <button
              className="login-button-letter"
              onClick={[Function]}
            >
              Create account
            </button>
            <button
              className="button-letter"
              type="submit"
            >
              Login
            </button>
          </div>
        </ValidatorForm>
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </div>,
]
`;
